frontend/src/hooks/useDragColumns/index.ts

```diff
@@ -0,0 +1,71 @@
+import { ColumnsType } from 'antd/es/table';
+import getFromLocalstorage from 'api/browser/localstorage/get';
+import setToLocalstorage from 'api/browser/localstorage/set';
+import { LOCALSTORAGE } from 'constants/localStorage';
+import useUrlQueryData from 'hooks/useUrlQueryData';
+import { useCallback, useEffect, useMemo } from 'react';
+
+import { COLUMNS } from './configs';
+
+const useDragColumns = <T>(storageKey: LOCALSTORAGE): UseDragColumns<T> => {
+	const {
+		query: draggedColumnsQuery,
+		queryData: draggedColumns,
+		redirectWithQuery: redirectWithDraggedColumns,
+	} = useUrlQueryData<ColumnsType<T>>(COLUMNS, []);
+
+	const localStorageDraggedColumns = useMemo(
+		() => getFromLocalstorage(storageKey),
+		[storageKey],
+	);
+
+	const handleRedirectWithDraggedColumns = useCallback(
+		(columns: ColumnsType<T>) => {
+			redirectWithDraggedColumns(columns);
+
+			setToLocalstorage(storageKey, JSON.stringify(columns));
+		},
+		[storageKey, redirectWithDraggedColumns],
+	);
+
+	const onDragColumns = useCallback(
+		(columns: ColumnsType<T>, fromIndex: number, toIndex: number): void => {
+			const columnsData = [...columns];
+			const item = columnsData.splice(fromIndex, 1)[0];
+			columnsData.splice(toIndex, 0, item);
+
+			handleRedirectWithDraggedColumns(columnsData);
+		},
+		[handleRedirectWithDraggedColumns],
+	);
+
+	useEffect(() => {
+		if (draggedColumnsQuery) return;
+
+		const nextDraggedColumns = localStorageDraggedColumns
```

Comment:
error handling is missing here

Reply:
@palashgdev  what type of error handling are you referring to? because in the current logic, if we don't have a url with columns, we'll use columns from local storage, and if we don't have columns in local storage, we just return an empty array, and it can't be affected

Reply:
**frontend/src/api/browser/localstorage/get.ts** it's handled here

Reply:
oh great and about JSON.parse also